### ISSUE TITLE
DnsClient: There is no option to enable the netty logger

### ISSUE THREAD

**Issue by stephanebastian** [eclipse/vert.x#2445]

**Comment by stephanebastian** [comment]
Sure. In fact I already created a branch for it in my repo. Still need to add a test to make sure that the netty logger is really enabled. 
BTW I just realized that there is no test for the 'property' logActivity on the various other client options (DatagramSocketOptions, EventBusOptions, HttpClientOpitons, etc.). Is it expected ?

For this PR, I am planning on creating a new class, say NettyLoggerTest, and call InternalLoggerFactory.setDefaultFactory(...) in the test to assert that we really got log messages. Is it ok for you if I use InternalLoggerFactory ?

**Comment by vietj** [comment]
@stephanebastian yes, I think by the time we added this feature we did not have logging testing. Now we do have some (in integration tests).

I think it's good to have this tested, however, I would like it to be part of integration tests instead like others. Look at JULLogDelegateTest, Log4J2LogDelegateTest, etc...

**Comment by stephanebastian** [comment]
@vietj I just pushed a PR for this issue. 
I am not sure I understood what you meant with the integration tests for logging. I just wanted to provide a test to make sure that the new property DnsClientOpitons.logActivity does what it's supposed to. Therefore I added two unit tests. Let me know if you want me to change this.
Also, I slightly modified the constructor of the class DnsClientImpl. It now only takes one arg (DnsClientOptions). This will ease the next PR that'll add another option to specify if we want the DNS server to do recursion or not
